src/client.ts

```diff
@@ -12,20 +12,102 @@ const extractJSONContent = /(\{[^]*\})/;
 
 const extractCQLInclude = /include .* called (.*)/g;
 
-export interface CqlObject {
-  main: string;
-  libraries: Library;
+export interface CqlLibraries {
```

Comment:
I notice these interfaces are exported, but I didn't see them used anywhere else.  Does it make sense to put these interfaces into their own file (maybe a `d.ts` file). Might be more natural to import them in dependent libraries that way.

Reply:
yea that sounds good, do I just specify the file path in the `package.json`?

Reply:
I haven't worked through it myself, but I think we might want

`declare module 'cql-translation-service-client'`

That way a client application could import it like:

`import { CqlLibraries, ElmLibraries, ElmLibrary } from 'cql-translation-service-client';`

[These were the docs I was looking at](https://www.typescriptlang.org/docs/handbook/modules.html#ambient-modules)

Reply:
So actually it looks like we might have to update the `package.json` for this like you said.

[TypeScript docs on publishing](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#including-declarations-in-your-npm-package)

Reply:
This should work. In hindsight my original suggestion wasn't really necessary (still might be nice to have types in their own file, but it has no impact on how projects using it as a dependency import them).

Looks like package.json is already configured to export the types properly.

If you `yarn build` on master, you'll get a `client.d.ts` file:

```typescript
export interface CqlObject {
    main: string;
    libraries: Library;
}
export interface Library {
    [name: string]: string;
}
export interface ElmObject {
    main: object;
    libraries: {
        [key: string]: object;
    };
}
export declare class Client {
    protected url: string;
    protected headers: object;
    constructor(serviceUrl: string);
    /**
     * Function that requests web_service to convert the cql into elm.
     * @param cql - cql file that is the input to the function.
     * @return The resulting elm translation of the cql file.
     */
    convertCQL(cql: CqlObject): Promise<ElmObject>;
    convertBasicCQL(cql: string): Promise<object>;
    private handleElm;
}
```

and on this branch a `client.d.ts` file::


```typescript
import { CqlLibraries, ElmLibraries, ElmLibrary } from 'types';
export declare class Client {
    protected url: string;
    protected headers: object;
    constructor(serviceUrl: string);
    /**
     * Function that requests web_service to convert the cql into elm.
     * @param cqlLibraries - object containing CqlLibraries that is the input to the function.
     * @return The resulting elm translation of the cql libraries.
     */
    convertCQL(cqlLibraries: CqlLibraries): Promise<ElmLibraries>;
    convertBasicCQL(cql: string): Promise<ElmLibrary>;
    private handleElm;
}
```

So I think we're good here.

Reply:
yea I couldn't figure out the right typescript config to get this working, I'll reverse this and just put all of the types in the client class.